src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerChangeService.java

```diff
@@ -422,9 +422,9 @@ private <P extends Psc> boolean setCommonAttributes(
           Boolean serviceAddressSameAsResidentialAddress
   ) {
 
-    var hasChange = changeManager.compareAndBuildRightChange(
+    var hasChange = changeManager.compareAndBuildLeftChange(
         Boolean.TRUE.equals(serviceAddressSameAsResidentialAddress) ? residentialAddress : serviceAddress,
-        PrivateBoDataApi::getPrincipalAddress,
+        PscApi::getPrincipalOfficeAddress,
```

Comment:
PscApi::getAddress()
[I now realise the change is for serviceAddress so can get from Public API]

Reply:
Cool I'll change that now